### PR TITLE
PCP-1467: UserAssignedIdentities Fix

### PR DESCRIPTION
**Pull request by vishu2498**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: azure/services/managedclusters/spec.go [hunk 1]
```diff
@@ -490,6 +502,14 @@ func computeDiffOfNormalizedClusters(managedCluster containerservice.ManagedClus
...
 
+	if managedCluster.Identity != nil {
+		clusterNormalized.Identity = managedCluster.Identity
+	}
+
+	if existingMC.Identity != nil {
+		existingMCClusterNormalized.Identity = existingMC.Identity
+	}
+
 	diff := cmp.Diff(clusterNormalized, existingMCClusterNormalized)
```
Comment (LochanRn) [inline azure/services/managedclusters/spec.go:516]: Quick doubt: Within the Identity struct it has readOnly variables, So ideally i am assuming azure service will auto-assign it. If it auto-assigns it existingMC.Identity and managedCluster.Identity will always be different. 

Did you verify if the diff is returned true only for the correct cases ?

Please verify the above case, also my suggestion would be to iterate over the map and assign uai alone
Something like below 

```
		uaIDs := make(map[string]*containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue)
		for key := range existingMC.Identity.UserAssignedIdentities {
			uaIDs[key] = &containerservice.ManagedClusterIdentityUserAssignedIdentitiesValue{
				// intentionally empty
			}
		}
		existingMCClusterNormalized.Identity = &containerservice.ManagedClusterIdentity{
			UserAssignedIdentities: uaIDs,
		}
```
Same for the clusterNormalized.Identity.  

We have seen in the past, if the diff is returned true for false case, the reconciler will bombard the azure server and the AKS cluster can get corrupted for the same reason.

File: azure/services/managedclusters/spec.go [hunk 2]
```diff
@@ -365,6 +363,20 @@ func (s *ManagedClusterSpec) Parameters(existing interface{}) (params interface{
 			}
 			existingMC.AddonProfiles[key].Identity = nil
 		}
+
+		if s.UserAssignedIdentities != nil {
```
Comment (LochanRn) [inline azure/services/managedclusters/spec.go]: The if condition here is redundant code to the line number 330 which is also doing the same.

File: azure/services/managedclusters/spec.go (hunk 1 above)
Comment (vishu2498) [reply azure/services/managedclusters/spec.go:516]: Verified all the cases. Also changed the code accordingly.
Thanks for the suggestion.

File: azure/services/managedclusters/spec.go (hunk 2 above)
Comment (vishu2498) [reply azure/services/managedclusters/spec.go]: Removed this code.

**Pull request merged**